service/type.go

```diff
@@ -161,3 +181,30 @@ type Parameter struct {
 	// Definition of the structure of the object when the type is object
 	Object []*Parameter `hash:"name:7" validate:"unique,dive,required"`
 }
+
+// Workflow describes a workflow of a service
+type Workflow struct {
+	Trigger *WorkflowTrigger `hash:"name:1" validate:"required"`
+	Task    *WorkflowTask    `hash:"name:2" validate:"required"`
+}
+
+// WorkflowTask describes the instructions for the workflow to execute a task
+type WorkflowTask struct {
+	InstanceHash hash.Hash `hash:"name:1" validate:"required"`
+	TaskKey      string    `hash:"name:2" validate:"printascii"`
+}
+
+// WorkflowTrigger is an event that triggers a workflow
+type WorkflowTrigger struct {
+	InstanceHash hash.Hash                `hash:"name:1" validate:"required"`
+	Key          string                   `hash:"name:2" validate:"printascii,printascii"`
+	Type         TriggerType              `hash:"name:3"`
```

Comment:
the `required` validation doesn't work I guess because TriggerType can be 0. It could be good to add validation tho if you have any idea

Reply:
You need to set a not used default value on the enum.
Something like:
```
const (
        UNKNOWNTRIGGERTYPE TriggerType = iota
	EVENT
	RESULT
)
```
https://github.com/mesg-foundation/engine/pull/1188/files#diff-fd2977ccac84eb22a5975801a69cf64cR40

Reply:
yes but with that, we lose the property that by default we are listening to events.
I could use the `oneof` validation but then it would be `oneof=0 1` which might be a bit weird. Or we could just keep it like that.

Reply:
why do you want to listen to events by default?

Reply:
It seems to me like a good default, if you put nothing it's because you want to react to an event but i'm ok not to have any default and have an error if not set and this will resolve the required part. I will do the fixes

Reply:
done